src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -48,14 +48,15 @@ public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObje
     /**
      * @description Batch size for OPPORTUNITY job mode
      */
-    public static final Integer OPPORTUNITIES_BATCH_SIZE = 2000;
+    public static final Integer OPPORTUNITIES_BATCH_SIZE = 200;
```

Comment:
I don't think this needs to be 200. You're never going to query more than 2 payment records per Opportunity, so even if every Opp in a batch of 2000 had 2 payments that would only be 4000 rows out of a max of 50000 rows.